cohort-engine/src/test/java/com/ibm/cohort/engine/measure/MeasureReportParameterHelperTest.java

```diff
@@ -0,0 +1,295 @@
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.DateTimeType;
+import org.hl7.fhir.r4.model.DateType;
+import org.hl7.fhir.r4.model.DecimalType;
+import org.hl7.fhir.r4.model.IntegerType;
+import org.hl7.fhir.r4.model.Period;
+import org.hl7.fhir.r4.model.Range;
+import org.hl7.fhir.r4.model.StringType;
+import org.hl7.fhir.r4.model.TimeType;
+import org.junit.Assert;
+import org.junit.Test;
+import org.opencds.cqf.cql.engine.runtime.Code;
+import org.opencds.cqf.cql.engine.runtime.Concept;
+import org.opencds.cqf.cql.engine.runtime.CqlList;
+import org.opencds.cqf.cql.engine.runtime.Date;
+import org.opencds.cqf.cql.engine.runtime.DateTime;
+import org.opencds.cqf.cql.engine.runtime.Interval;
+import org.opencds.cqf.cql.engine.runtime.Quantity;
+import org.opencds.cqf.cql.engine.runtime.Ratio;
+import org.opencds.cqf.cql.engine.runtime.Time;
+
+public class MeasureReportParameterHelperTest {
+
+	@Test
+	public void testUnsupportedType_shouldReturnNull() {
+		assertNull(MeasureReportParameterHelper.getFhirTypeValue(new CqlList()));
+	}
+
+	@Test
+	public void testIntervalDateTime_shouldReturnPeriodInUTC() {
+		DateTime startDateTime = new DateTime(OffsetDateTime.of(LocalDateTime.of(2020, 1, 1, 0, 10, 5, 0), ZoneOffset.of("Z")));
+		DateTime endDateTime = new DateTime(OffsetDateTime.of(LocalDateTime.of(2020, 2, 15, 15, 10, 0, 0), ZoneOffset.of("+01:00")));
+
+		Interval interval = new Interval(startDateTime, true, endDateTime, true);
+
+		IBaseDatatype fhirTypeValue = MeasureReportParameterHelper.getFhirTypeValue(interval);
+
+		// Expect output expressed in UTC
+		DateTimeType expectedStart = new DateTimeType("2020-01-01T00:10:05+00:00");
+		DateTimeType expectedEnd = new DateTimeType("2020-02-15T14:10:00+00:00");
+
+		assertTrue(fhirTypeValue instanceof Period);
+		Period castResult = (Period) fhirTypeValue;
+		assertEquals(expectedStart.toHumanDisplay(), castResult.getStartElement().toHumanDisplay());
+		assertEquals(expectedEnd.toHumanDisplay(), castResult.getEndElement().toHumanDisplay());
+	}
+
+	@Test
+	public void testIntervalQuantity_shouldReturnRange() {
+		String unit = "ml";
+		BigDecimal startAmount = new BigDecimal("4.0");
+		BigDecimal endAmount = new BigDecimal("5.0");
+		Quantity startQuantity = new Quantity().withUnit(unit).withValue(startAmount);
+		Quantity endQuantity = new Quantity().withUnit(unit).withValue(endAmount);
+
+		IBaseDatatype fhirTypeValue = MeasureReportParameterHelper.getFhirTypeValue(new Interval(startQuantity, true, endQuantity, true));
+
+		assertTrue(fhirTypeValue instanceof Range);
+		Range castResult = (Range) fhirTypeValue;
+		verifyBaseTypeAsQuantity(castResult.getLow(), startAmount, unit);
+		verifyBaseTypeAsQuantity(castResult.getHigh(), endAmount, unit);
+	}
+
+	/*
+	 *  Currently only supporting CQL intervals of type DateTime or Quantity.
+	 *  Support for intervals of other types may be added in the future.
+	 */
```

Comment:
This comment feels like it is old/out of date. Is it still relevant?

Reply:
It should still be true.

Reply:
I see it now. You have tests for other types, but they all test for null parameter output.